test/stripe/api_resource_test.rb

```diff
@@ -274,6 +313,34 @@ class NestedTestAPIResource < Stripe::APIResource
         assert_equal false, c.livemode
       end
 
+      should "putting api_key correctly as an opt should not trigger a warning" do
+        stub_request(:post, "#{Stripe.api_base}/v1/customers").
+          to_return(body: JSON.generate(API_FIXTURES.fetch(:customer)))
+        c = Stripe::Customer.new
+        old_stderr = $stderr
+        $stderr = StringIO.new
+        begin
+          c.save({}, { :api_key => 'sk_test_local' })
+          assert $stderr.string.empty?
+        ensure
+          $stderr = old_stderr
+        end
+      end
+
+      should "putting api_key as a param instead of an opt when updating an object should trigger a warning" do
+        stub_request(:post, "#{Stripe.api_base}/v1/customers").
+          to_return(body: JSON.generate(API_FIXTURES.fetch(:customer)))
+        c = Stripe::Customer.new
+        old_stderr = $stderr
+        $stderr = StringIO.new
+        begin
+          c.save({ :api_key => 'sk_test_local' })
+          assert_match Regexp.new('WARNING:'), $stderr.string
+        ensure
+          $stderr = old_stderr
+        end
+      end
```

Comment:
Good thinking, but I think having tests for both idempotency key and API key for create/update is a bit overkill. Let's take these two out okay?

Maybe you can generalize the name of the other with something like just refer to a general "opt" instead of `api_key` or `idempotency_key` specifically.